components/calculation/PopulationResultsTable.tsx

```diff
@@ -0,0 +1,93 @@
+import { Table } from '@mantine/core';
+import { useMemo } from 'react';
+import { DetailedResult } from '../../util/types';
+
+export type PopulationResultViewerProps = {
+  results: DetailedResultInfoArray;
+};
+
+export type DetailedResultInfoArray = Array<DetailedDetailedResult>;
+
+export declare type DetailedDetailedResult = {
+  label: string;
+  detailedResult: DetailedResult;
+};
+
+export type PopulationResult = {
+  label: string;
+  [key: string]: number | string;
+};
+
+export default function PopulationResultTable({ results }: PopulationResultViewerProps) {
+  const tableHeaders = useMemo(() => {
+    return extractTableHeaders(results?.[0].detailedResult);
+  }, [results]);
+
+  /**
+   * Converts an array of DetailedResults with labels and formats it into JSX table rows for display
+   */
+  function constructPopulationResultsArray(results: DetailedResultInfoArray) {
+    return results
+      .filter(drInfo => drInfo.detailedResult)
+      .map(drInfo => {
+        return extractPopulationResultRow(drInfo.detailedResult, drInfo.label);
+      });
+  }
+
+  /**
+   * Strips population results off of a DetailedResult and calls PopulationResultsRow to format them into a TSX component
+   */
+  function extractPopulationResultRow(dr: DetailedResult, label = 'Unlabeled Patient') {
+    const group = dr.detailedResults?.[0];
+    const populationResults = { label };
+    group?.populationResults?.reduce((acc: PopulationResult, e) => {
+      const key = e.criteriaExpression || e.populationType;
+      if (key) {
+        acc[key as string] = e.result === true ? 1 : 0;
+      }
+      return acc;
+    }, populationResults);
+    return PopulationResultsRow(populationResults);
```

Comment:
The term `populationResults` is super overloaded here, consider changing the variable to `labeledPopulationResults` or something?